src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/page/DashboardPage.kt

```diff
@@ -14,10 +14,17 @@ class DashboardPage(
     private val jiraErrors = JiraErrors(driver)
 
     fun dismissAllPopups() {
-        driver.findElements(By.cssSelector(".aui-flag .icon-close")).filter { it.isEnabled && it.isDisplayed }.forEach { it.click() }
-        driver.findElements(By.id("nps-acknowledgement-accept-button")).filter { it.isEnabled && it.isDisplayed }.forEach { it.click() }
-        driver.findElements(By.cssSelector(".jira-help-tip .cancel")).filter { it.isEnabled && it.isDisplayed }.forEach { it.click() }
-        driver.findElements(By.className("postsetup-close-link")).filter { it.isEnabled && it.isDisplayed }.forEach { it.click() }
+        val popupDismissalLocators = listOf(
+            By.id("nps-acknowledgement-accept-button"),
+            By.cssSelector(".jira-help-tip .cancel"),
+            By.className("postsetup-close-link")
+        )
+        
+        for (locator in popupDismissalLocators) {
+            driver.findElements(locator)
+                .filter { it.isEnabled && it.isDisplayed }
+                .forEach { it.click() }
+        }
```

Comment:
If you want to keep chaining:
```kotlin
listOf(
    By.id("nps-acknowledgement-accept-button"),
    By.cssSelector(".jira-help-tip .cancel"),
    By.className("postsetup-close-link")
)
    .flatMap { driver.findElements(it) }
    .filter { it.isEnabled && it.isDisplayed }
    .forEach { it.click() }
```

Reply:
I don't, really - there may be an overlap between these selectors, for example AUI flags can be handled by the "close" button but also "Don't remind me again". If we run the second first, the first one won't be interactable anymore and will result in an exception.

Reply:
`listOf().asSequence()` should do the trick